src/types/array.js

```diff
@@ -4,9 +4,15 @@ import ValidationError from '../lib/error'
 
 const array = {
   default: context => {
+    // Ensure array
     if (!Array.isArray(context.value)) {
       return context.fail('Value must be an array')
     }
+    // If empty (and empty allowed), move forward
+    if (context.def.empty && context.value.length === 0) {
+      return context.value
+    }
```

Comment:
We're not failing in the event that empty isn't allowed, and context.value is an empty array